lib/chef-dk/command/verify.rb

```diff
@@ -141,6 +147,35 @@ def components
           # ------------
           failures = []
 
+          # fail the verify if we have more than one version of chef-provisioning or any of its drivers.
```

Comment:
It's also possible folks will have done `chef gem install chef-provisioning-foo` which will leave them with another version in their home directory. We should either account for that in the test or add some output to the test to indicate that it's only a valid test on a clean machine.